src/metabase/sync/sync_metadata.clj

```diff
@@ -49,6 +49,8 @@
    (sync-util/create-sync-step "sync-fields" sync-fields/sync-fields! sync-fields-summary)
    ;; Now for each table, sync the FKS. This has to be done after syncing all the fields to make sure target fields exist
    (sync-util/create-sync-step "sync-fks" sync-fks/sync-fks! sync-fks-summary)
+   ;; Now set initial_sync_status=complete for each table
+   (sync-util/create-sync-step "set-initial-table-sync-complete" sync-util/set-all-tables-initial-sync-status-complete!)
```

Comment:
This new step is effectively replacing [this](https://github.com/metabase/metabase/pull/31607/files#r1231041004) the calls to `sync-util/set-initial-table-sync-complete!` for each table. The biggest difference is doing it this way requires only one update query for all tables instead of one update per table

Reply:
are you sure this is what we want?
This changes the behavior of complete sync from gradually to "all at once".

there was a discussion about being able to play around with a table as soon as it's finished syncing, I imagine with this change we can only play with a table when the whole db is done syncing. Is my assessment correct?

Reply:
You're right, we definitely want to keep that behaviour. I've restored it